plugins/modules/instance_rdns_record.py

```diff
@@ -0,0 +1,189 @@
+    content: www.example.com
+
+- name: Delete the reverse DNS for a compute instance
+  ngine_io.exoscale.instance_rdns_record:
+    name: web-vm-1
+    state: absent
+'''
+
+RETURN = '''
+---
+instance_rdns_domain:
+  description: Reverse DSN name of the compute instance
+  returned: success
+  type: str
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_bytes, to_text
+from ansible_collections.ngine_io.cloudstack.plugins.module_utils.cloudstack import (
+    AnsibleCloudStack,
+    cs_argument_spec,
+    cs_required_together
+)
+
+
+class InstanceRdnsRecord(AnsibleCloudStack):
+
+    def __init__(self, module):
+        super(InstanceRdnsRecord, self).__init__(module)
+
+        self.name = self.module.params.get('name')
+        self.content = self.module.params.get('content')
+
+        self.returns = {
+            'domain': 'domain'
+        }
+
+        self.instance = None
+
+    def get_instance(self):
+        instance = self.instance
+        if not instance:
+            args = {
+                'fetch_list': True,
+            }
+            # Do not pass zoneid, as the instance name must be unique across
+            # zones.
+            instances = self.query_api('listVirtualMachines', **args)
+            if instances:
+                for v in instances:
+                    if self.name.lower() in [v['name'].lower(), v['displayname'].lower(), v['id']]:
+                        self.instance = v
+                        break
+
+        return self.instance
+
+    def get_record(self, instance):
+        result = {}
+
+        res = self.query_api('queryReverseDnsForVirtualMachine',
+                             id=instance['id'])
+        nics = res['virtualmachine'].get('nic', [])
+
+        defaultnics = [nic for nic in nics if nic.get('isdefault', False)]
+
+        if len(defaultnics) > 0:
+            domains = [record['domainname'] for record
+                       in defaultnics[0].get('reversedns', [])
+                       if 'domainname' in record]
+
+            if len(domains) > 0:
+                result = {
+                    'domainname': domains[0],
+                }
+
+        return result
+
+    def present_record(self):
+        instance = self.get_instance()
+
+        if not instance:
+            self.module.fail_json(
+                msg="No compute instance with name=%s found. " % self.name)
+
+        data = {
+            'domainname': self.content,
+        }
+        record = self.get_record(instance)
+        if self.has_changed(data, record):
+            self.result['changed'] = True
+            self.result['diff']['before'] = record
+            self.result['diff']['after'] = data
+            if not self.module.check_mode:
+                self.query_api('updateReverseDnsForVirtualMachine',
+                               id=instance['id'],
+                               domainname=data['domainname'])
+
+        return data
```

Comment:
I would return what we get from the api.
~~~diff
-                self.query_api('updateReverseDnsForVirtualMachine',
-                              id=instance['id'],
-                              domainname=data['domainname'])

-        return data

+                record = self.query_api(
+                    'updateReverseDnsForVirtualMachine',
+                    id=instance['id'],
+                    domainname=data['domainname']
+                )
+                if 'id' in record: del record['id']

+        return record
~~~